prow/plugins/lgtm/lgtm.go

```diff
@@ -43,67 +58,146 @@ type githubClient interface {
+			rce.PullRequest.Number,
+		),
+		htmlurl: rce.Comment.HTMLURL,
+	}
+	return handle(pc.GitHubClient, pc.Logger, e)
+}
+
+func handle(gc githubClient, log *logrus.Entry, e *event) error {
+	if e.action != "created" && e.action != "submitted" {
+		return nil
+	}
 
 	// If we create an "/lgtm" comment, add lgtm if necessary.
 	// If we create a "/lgtm cancel" comment, remove lgtm if necessary.
 	wantLGTM := false
-	if lgtmRe.MatchString(ic.Comment.Body) {
+	if lgtmRe.MatchString(e.body) {
 		wantLGTM = true
-	} else if lgtmCancelRe.MatchString(ic.Comment.Body) {
+	} else if lgtmCancelRe.MatchString(e.body) {
 		wantLGTM = false
 	} else {
 		return nil
 	}
 
 	// Allow authors to cancel LGTM. Do not allow authors to LGTM, and do not
 	// accept commands from any other user.
-	commentAuthor := ic.Comment.User.Login
-	isAssignee := ic.Issue.IsAssignee(commentAuthor)
-	isAuthor := ic.Issue.IsAuthor(commentAuthor)
+	isAssignee := false
+	for _, user := range e.assignees {
+		if user.Login == e.commentAuthor {
+			isAssignee = true
+			break
+		}
+	}
+	isAuthor := e.commentAuthor == e.prAuthor
 	if isAuthor && wantLGTM {
 		resp := "you cannot LGTM your own PR"
 		log.Infof("Commenting with \"%s\".", resp)
-		return gc.CreateComment(org, repo, number, plugins.FormatICResponse(ic.Comment, resp))
+		return gc.CreateComment(e.org, e.repo, e.number, plugins.FormatResponseRaw(e.body, e.htmlurl, e.commentAuthor, resp))
 	} else if !isAuthor && !isAssignee {
-		log.Infof("Assigning %s/%s#%d to %s", org, repo, number, commentAuthor)
-		if err := gc.AssignIssue(org, repo, number, []string{commentAuthor}); err != nil {
+		log.Infof("Assigning %s/%s#%d to %s", e.org, e.repo, e.number, e.commentAuthor)
+		if err := gc.AssignIssue(e.org, e.repo, e.number, []string{e.commentAuthor}); err != nil {
 			msg := "assigning you to the PR failed"
-			if ok, merr := gc.IsMember(org, commentAuthor); merr == nil && !ok {
+			if ok, merr := gc.IsMember(e.org, e.commentAuthor); merr == nil && !ok {
 				msg = "only kubernetes org members may be assigned issues"
 			} else if merr != nil {
-				log.WithError(merr).Errorf("Failed IsMember(%s, %s)", org, commentAuthor)
+				log.WithError(merr).Errorf("Failed IsMember(%s, %s)", e.org, e.commentAuthor)
 			} else {
-				log.WithError(err).Errorf("Failed AssignIssue(%s, %s, %d, %s)", org, repo, number, commentAuthor)
+				log.WithError(err).Errorf("Failed AssignIssue(%s, %s, %d, %s)", e.org, e.repo, e.number, e.commentAuthor)
 			}
 			resp := "changing LGTM is restricted to assignees, and " + msg
 			log.Infof("Reply to assign via /lgtm request with comment: \"%s\"", resp)
-			return gc.CreateComment(org, repo, number, plugins.FormatICResponse(ic.Comment, resp))
+			return gc.CreateComment(e.org, e.repo, e.number, plugins.FormatResponseRaw(e.body, e.htmlurl, e.commentAuthor, resp))
 		}
 	}
 
 	// Only add the label if it doesn't have it, and vice versa.
-	hasLGTM := ic.Issue.HasLabel(lgtmLabel)
+	hasLGTM := e.hasLabel(lgtmLabel)
```

Comment:
I think this is done backwards. Have the `hasLabel` member return `(bool, error)` and handle it here.